### PR TITLE
Fixed LMMS crash when pressing Q in not existing piano roll.

### PR DESCRIPTION
**Pull request by karmux**
Steps to reproduce:
1. Open LMMS
2. Open Piano-Roll
3. Click on Q
4. LMMS crashes.

This change fixes it.
Link to issue: #3608.

### CODE REVIEW

File: src/gui/editors/PianoRoll.cpp [hunk 1]
```diff
@@ -3918,31 +3918,33 @@ int PianoRoll::quantization() const
 
 void PianoRoll::quantizeNotes()
 {
-	NoteVector notes = getSelectedNotes();
+	if( m_pattern != NULL ) {
```
Comment (michaelgregorius) [inline src/gui/editors/PianoRoll.cpp]: While this definitively works I'd propose to use the same style like in `PianoRoll::cutSelectedNotes()`:
```C++
if( ! hasValidPattern() )
{
	return;
}
```
This make it more obvious what's tested and also keeps the indentation level of the method down.

File: src/gui/editors/PianoRoll.cpp [hunk 2]
```diff
@@ -805,7 +805,7 @@ void PianoRoll::setBackgroundShade( const QColor & c )
 
 
 
-void PianoRoll::drawNoteRect( QPainter & p, int x, int y, 
+void PianoRoll::drawNoteRect( QPainter & p, int x, int y,
 				int width, const Note * n, const QColor & noteCol,
```
Comment (zonkmachine) [inline src/gui/editors/PianoRoll.cpp]: You have one requested fix in here together with 14 lines that are formatting changes like whitespace at the end etc. This makes it a bit hard to review. I suggest for the next time to commit changes like this separately and name the commit 'fixup' or 'whitespace' or something similar.

Comment (tresf) [inline src/gui/editors/PianoRoll.cpp]: @zonkmachine agreed.  Harder to cherry-pick too as the more lines that get modified increases the chances of a merge conflict.

Comment (karmux) [inline src/gui/editors/PianoRoll.cpp]: I need to switch off editor's automatic cleanups :)

File: src/gui/editors/PianoRoll.cpp [hunk 3]
```diff
@@ -3918,31 +3918,33 @@ int PianoRoll::quantization() const
 
 void PianoRoll::quantizeNotes()
 {
-	NoteVector notes = getSelectedNotes();
+	if( hasValidPattern() ) {
```
Comment (michaelgregorius) [inline src/gui/editors/PianoRoll.cpp]: This is already better than before but I was also asking to use the following coding pattern:
```C++
if ( !hasValidPattern() )
{
  return;
}

NoteVector notes = getSelectedNotes();

// ...
// Everything that's inside the `if` statement follows here
```
This is an often used pattern when the scope of an `if` statement encompasses a whole method. By doing it like this you also get rid of one indentation level for the code that's inside the `if` clause which makes the code more readable.

File: src/gui/editors/PianoRoll.cpp [hunk 4]
```diff
@@ -3918,33 +3918,36 @@ int PianoRoll::quantization() const
 
 void PianoRoll::quantizeNotes()
 {
-	if( hasValidPattern() ) {
-		NoteVector notes = getSelectedNotes();
+	if( ! hasValidPattern() )
```
Comment (michaelgregorius) [inline src/gui/editors/PianoRoll.cpp]: That's what I meant. Thanks!

Comment (zonkmachine) [inline src/gui/editors/PianoRoll.cpp]: Clean and simple. Retested, merging!

**Comment by tresf** [comment]
FYI, this commit is [already to merge to master](https://github.com/liushuyu/lmms/blob/merge/src/gui/editors/PianoRoll.cpp#L3921) in `liushuyu/merge`.

File: src/gui/editors/PianoRoll.cpp [hunk 5]
```diff
@@ -3918,27 +3918,32 @@ int PianoRoll::quantization() const
...
 {
+	if( ! hasValidPattern() )
+	{
+		return;
+	}
+
 	NoteVector notes = getSelectedNotes();
 
-	if (notes.empty())
+	if( notes.empty() )
```
Comment (grejppi) [inline src/gui/editors/PianoRoll.cpp]: This is a new function, and therefore the old coding style (wrt spaces inside parentheses) should not apply. I realize this is late, but still 😅

**Pull request merged**